lib/OpenQA/Shared/Controller/Session.pm

```diff
@@ -21,13 +21,14 @@ use Carp 'croak';
 sub ensure_user {
     my ($self) = @_;
     return 1 if $self->current_user;
-    $self->redirect_to('login');
+    $self->redirect_to($self->url_for('login')->query(return_page => $self->req->url));
```

Comment:
The exact stringified format of `$c->req->url` depends on how the server is deployed. In some cases it might only contain parts of the path (and a prefix would be stored in `$c->req->url->base`). I've not seen the webapi deployed that way yet though, so it's a theoretical issue. The safe way would be to use `$c->req->url->to_abs`, which guarantees the absolute URL format.

Reply:
I want a relative way anyway as there is no reason for an absolute url as the redirect from `/response` (or was it `/return`?) to the original page can be relative.
In fact `/login` only takes the `path_query`.